python/marvin/__init__.py

```diff
@@ -403,6 +403,14 @@ def _check_config(self):
         # update the allowed releases
         self._update_releases()
 
+        # If the configuration file included a default release, we set it now
```

Comment:
Shouldn't this only set the release on the initial import?  In its current location within `check_config`, it will also get called and set on every call to `config.access`.  Is this the behaviour you're going for?

Reply:
No, not really. I modified it so that it sets `self._release` in `_load_defaults`. That seems to have the desired effect.

There is a bit of a frail design in `_replant_tree` in which it can fail because the tree has not yet been set at that point. I had to do a bit of hacking (see line 423 and comment). I think it is ok, though.

Reply:
Hmm.  I'll have a look.  `_replant_tree` should do nothing and yield the value cleanly if no tree is set or is nothing has changed regarding `releases` or `access` settings.  That's the desired behaviour.